audiorecorder/src/main/java/org/odk/collect/audiorecorder/recording/AudioRecorderService.kt

```diff
@@ -0,0 +1,98 @@
+package org.odk.collect.audiorecorder.recording
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.Service
+import android.content.Intent
+import android.os.Build
+import android.os.IBinder
+import androidx.core.app.NotificationCompat
+import org.odk.collect.audiorecorder.R
+import org.odk.collect.audiorecorder.getComponent
+import org.odk.collect.audiorecorder.recorder.Recorder
+import org.odk.collect.strings.getLocalizedString
+import javax.inject.Inject
+
+class AudioRecorderService : Service() {
+
+    @Inject
+    internal lateinit var recorder: Recorder
+
+    @Inject
+    internal lateinit var recordingRepository: RecordingRepository
+
+    override fun onCreate() {
+        super.onCreate()
+        getComponent().inject(this)
+    }
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        when (intent?.action) {
+            ACTION_START -> {
+                setupNotificationChannel()
+
+                val notification = NotificationCompat.Builder(this, NOTIFICATION_CHANNEL)
+                    .setContentTitle(getLocalizedString(R.string.recording))
+                    .setSmallIcon(R.drawable.ic_baseline_mic_24)
+                    .build()
+
+                startForeground(NOTIFICATION_ID, notification)
+
+                if (!recorder.isRecording()) {
+                    recorder.start()
+                }
+            }
+
+            ACTION_CANCEL -> {
+                cancelRecording()
+            }
+
+            ACTION_STOP -> {
+                stopRecording()
+            }
+        }
+
+        return START_STICKY
+    }
+
+    private fun setupNotificationChannel() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationChannel = NotificationChannel(
+                NOTIFICATION_CHANNEL,
+                getLocalizedString(R.string.recording_channel),
```

Comment:
Had to think about this one. I briefly wondered whether there might be a more general channel name that would make sense but I think this is the way to go.